pages/agent/v3.md

```diff
@@ -165,6 +136,56 @@ The job API is unavailable on agents running versions of Windows before build 17
 >🛠 Experimental feature
 > To use the job API, set <code>experiment="job-api"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a>.
 
+### Agent API
+
+Like the Job API experiment above, this exposes a (separate) local API for interacting with the agent process.
+The Agent API offers these endpoints:
+
+- `GET /api/leader/v0/ping` - Returns a JSON object with the current time (useful for testing the agent is alive).
+- `GET /api/leader/v0/lock?key=<key>` - Returns a JSON object containing the current state of a lock.
+- `PATCH /api/leader/v0/lock?key=<key>` - Accepts a JSON object with old and new states for a lock. The lock is then updated atomically, and a JSON object describing whether the operation proceeded is returned
+
+The API is exposed via a Unix Domain Socket. Unlike the `job-api`, the path to the socket is not available via a environment variable - rather, there is a single (configurable) path on the system.
+
+>🛠 Experimental feature
+> To use the agent API, set <code>experiment="agent-api"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a>.
+
+
+## Promoted experiments
+
+We've recently promoted some experiments into full features.
```

Comment:
This triggered a couple of things for me:

- Time words like "recently" are tricky in docs so it's good to avoid them.
- This feels like changelog information rather than docs.

BUT, it seems like useful information. Is there anywhere else it could go? Or perhaps just a different heading that doesn't tie it to a time?

Reply:
Actually, mentioning the relevant versions like you have, helps. Have we gone through this lifecycle with other experiments? If so, how did they get handled?

Reply:
Good question. I thought this might be the first time we've actually "completed" experiments. But it turns out that there is at least one completed experiment present in this file - `output-redactor` (Redacted variables) was de-experimented in mid 2021 (see https://github.com/buildkite/agent/pull/1452). 

Past experiments feels like important documentation to have ("why is my agent printing 'The ansi-timestamps experiment has been promoted...'?") and this seemed like the right page to put it. Perhaps it should be a subsection of "Experimental features"?

Reply:
I've left a suggestion for handling this in #2240. Let me know what you think there.